scri/asymptotic_bondi_data/map_to_abd_frame.py

```diff
@@ -0,0 +1,478 @@
+import numpy as np
+
+import sxs
+import scri
+import spinsfast
+import spherical_functions as sf
+from spherical_functions import LM_index
+
+import quaternion
+from quaternion.calculus import derivative
+from quaternion.calculus import indefinite_integral as integrate
+
+from scipy.interpolate import CubicSpline
+
+from functools import partial
+
+from scri.asymptotic_bondi_data.map_to_superrest_frame import time_translation, rotation
+
+
+def cost(δt_δϕ, args):
+    modes_A, modes_B, t_reference, δϕ_factor, δΨ_factor, normalization = args
+
+    # Take the sqrt because least_squares squares the inputs...
+    diff = integrate(
+        np.sum(
+            abs(modes_A(t_reference + δt_δϕ[0]) * np.exp(1j * δt_δϕ[1]) ** δϕ_factor * δΨ_factor - modes_B) ** 2, axis=1
+        ),
+        t_reference,
+    )[-1]
+    return np.sqrt(diff / normalization)
+
+
+def align2d(h_A, h_B, t1, t2, n_brute_force_δt=None, n_brute_force_δϕ=None, include_modes=None, nprocs=4):
```

Comment:
Ah, I remember now.  Another neglected PR to sxs has similar code.  If I merge that, could we just reuse it here?

Reply:
I just realized that scri and sxs depend on each other... is this ok?

Reply:
`sxs` doesn't *really* depend on `scri`; it's optional, and I try to never use `scri` in `sxs`.  The reason is just that `scri` can be so hard to install, and `sxs` is intended to be this simple, pure-python package.  I think the only use is in [`version_info`](https://github.com/sxs-collaboration/sxs/blob/8f46c1c1c9c6de8ac69b69564d6a538934610547/sxs/utilities/__init__.py#L28), where it just sees if `scri` can be loaded, so it can record its version number; if not, it just skips it.  This sort of thing is fine in python, as long as you're not creating circular imports by putting `import scri` and `import sxs` at the top levels.  Basically, as long as one is hidden in a function, it's no problem.

On the other hand, I'm perfectly happy to use `sxs` from `scri`, and there are already instances of that.

Reply:
Yeah I'd be fine with calling `sxs` to do the 2d alignment. Just lmk when that code is merged into `scri`!